src/ILCompiler.Compiler/src/Compiler/MetadataManager.cs

```diff
@@ -42,15 +42,15 @@ public abstract class MetadataManager
         private HashSet<GenericDictionaryNode> _genericDictionariesGenerated = new HashSet<GenericDictionaryNode>();
         private List<TypeGVMEntriesNode> _typeGVMEntries = new List<TypeGVMEntriesNode>();
 
-        private Dictionary<DynamicInvokeMethodSignature, MethodDesc> _dynamicInvokeThunks = new Dictionary<DynamicInvokeMethodSignature, MethodDesc>();
-
         internal NativeLayoutInfoNode NativeLayoutInfo { get; private set; }
 
         public MetadataManager(NodeFactory factory)
         {
             _nodeFactory = factory;
         }
 
+        public NodeFactory NodeFactory { get { return _nodeFactory; } }
```

Comment:
Can we make `_nodeFactory` a protected readonly field instead? Or do we need `MetadataManager` to expose this publically?